server/verify.go

```diff
@@ -0,0 +1,84 @@
+package server
+
+import (
+	"bytes"
+	"crypto"
+	"crypto/rsa"
+	"crypto/subtle"
+	"fmt"
+	"hash"
+	"sort"
+
+	"github.com/google/go-tpm-tools/proto"
+	"github.com/google/go-tpm-tools/tpm2tools"
+	"github.com/google/go-tpm/tpm2"
+)
+
+// Verify will take in a public key (currently, only RSA is supported), a proto
+// (contains an attestation and a signature), and will check the attestation is
+// valid or not.
+// In addition to the basic RSA signature checking, this function will also check
+// the PCR digest for quoted data, which can insure that the given PCR state is the
+// same as when signing this attestation. This function will return the "Extra Data"
+// appended in the attestation when quoting if passed all verification.
+func Verify(pubkey crypto.PublicKey, quote *proto.Quote, pcrs *proto.Pcrs) ([]byte, error) {
+	sig, err := tpm2.DecodeSignature(bytes.NewBuffer(quote.GetRawSig()))
+	if err != nil {
+		return nil, fmt.Errorf("signature decoding failed: %v", err)
+	}
+	if rsaPubKey, ok := pubkey.(*rsa.PublicKey); ok {
+		if sig.Alg != tpm2.AlgRSASSA && sig.Alg != tpm2.AlgRSAPSS {
+			return nil, fmt.Errorf("given signature is not a RSA signature: %v", sig.Alg)
+		}
+
+		var attestationHash hash.Hash
+		if sig.RSA.HashAlg == tpm2.AlgSHA256 {
```

Comment:
Should we be hardcoding this hash algo?

Reply:
You mean we should support all RSA hash algorithms?
I only added sha256, because I think all tpm 2.0 default key templates have sha256 as hashalgo.

Reply:
Yes we need, I over looked this part..

Reply:
You mean we should support all RSA hash algorithms?
I only added sha256, because I think all tpm 2.0 default key templates have sha256 as hashalgo.

Reply:
We should (at least initially) just support whatever hash algorithm we get with sig.RSA.HashAlg. This is what we do in other places as well

Reply:
Yes currently this only support SHA256 which is used by the default template , otherwise it will throw exception